### PR TITLE
[SW-572] Fetch active multisig transactions

### PR DESCRIPTION
**Pull request by subhod-i**

### CODE REVIEW

File: src/composables/multisig.ts [hunk 1]
```diff
@@ -76,8 +89,51 @@ export function useMultisig({ store }: UseMultisigOptions) {
...
+        confirmationsRequired: txConsensus.confirmations_required,
+        confirmedBy: txConsensus.confirmed_by,
+        hasConsensus: txConsensus.hasConsensus,
+        hash,
+        tx: rawTx ? (TxBuilder.unpackTx(rawTx.tx)).tx : null,
+      };
+    }
+  }
+
+  initPollingWatcher(() => fetchActiveMultisigTx('ct_M9XmoVp1Jq1yzpzYb57xv8qLh7UoKHooMVJH9PVnXU1RFboVf', 'ak_LbAJdCyErqU8mU3bPCbasmPoqm8EvC9Cfa8Ztf7K69QNphhzG'), POLLING_INTERVAL);
```
Comment (subhod-i) [inline src/composables/multisig.ts:131]: Kept this hardcoded values for easier integration.

File: src/composables/multisig.ts [hunk 2]
```diff
@@ -13,13 +20,19 @@ interface UseMultisigOptions {
...
+
 /**
  * This composable provides methods related to multisig operations
  */
 export function useMultisig({ store }: UseMultisigOptions) {
   const { getDrySdk, getSdk } = useSdk({ store });
+  const activeNetwork = computed<INetwork>(() => store.getters.activeNetwork);
 
   const multisigProgress = ref<IMultisigCreationStep>(MULTISIG_CREATION_STEPS.preparing);
+  const activeMultisigTransactions = ref<Record<string, IActiveMultisigTx>>({});
```
Comment (subhod-i) [inline src/composables/multisig.ts:33]: A  map where the key is the multisig account Id and the value is the active transaction with all the consensus statuses. This transaction detail could be merged with the unified multisig account data structure too.

File: src/composables/multisigTransactions.ts [hunk 3]
```diff
@@ -11,7 +11,8 @@ import {
   postJson,
 } from '../popup/utils';
 import type {
-  INetwork,
+  IActiveMultisigTx,
+  INetwork, IRawMultisigTx,
```
Comment (peronczyk) [inline src/composables/multisigTransactions.ts:15]: I suggest to put all items in one line or all in separate.

File: src/types/index.ts [hunk 4]
```diff
@@ -448,3 +448,19 @@ export type IMultisigCreationStep = keyof typeof MULTISIG_CREATION_STEPS;
 export interface ICreateMultisigAccount {
   address: string
 }
+export interface IActiveMultisigTx {
```
Comment (peronczyk) [inline src/types/index.ts:451]: Please add empty line before this interface.

File: src/types/index.ts [hunk 5]
```diff
@@ -448,3 +448,19 @@ export type IMultisigCreationStep = keyof typeof MULTISIG_CREATION_STEPS;
...
+  totalConfirmations: number,
+  confirmationsRequired: number,
+  confirmedBy: string[],
+  hasConsensus: boolean,
+  hash: string | undefined,
+  tx?: ITx,
```
Comment (peronczyk) [inline src/types/index.ts:457]: No need to add commas at the end.

File: src/types/index.ts [hunk 6]
```diff
@@ -448,3 +448,19 @@ export type IMultisigCreationStep = keyof typeof MULTISIG_CREATION_STEPS;
...
+}
+
+export interface IRawMultisigTx {
+  id: number;
+  hash: string;
+  tx: string;
+  createdAt: Date;
+  updatedAt: Date;
```
Comment (peronczyk) [inline src/types/index.ts:465]: No need to add semicolons at the end of lines.

File: src/types/index.ts [hunk 7]
```diff
@@ -448,3 +448,19 @@ export type IMultisigCreationStep = keyof typeof MULTISIG_CREATION_STEPS;
 export interface ICreateMultisigAccount {
   address: string
 }
+export interface IActiveMultisigTx {
+  totalConfirmations: number,
+  confirmationsRequired: number,
+  confirmedBy: string[],
+  hasConsensus: boolean,
+  hash: string | undefined,
```
Comment (peronczyk) [inline src/types/index.ts:456]: If it's possible that the value is undefined use question mark:
```
hash?: string
```

File: src/composables/multisigTransactions.ts [hunk 8]
```diff
@@ -42,8 +43,44 @@ export function useMultisigTransactions({ store }: UseMultisigTransactionsOption
...
+      .then((res) => res)
+      .catch(null);
+  }
+
+  /**
+ * Get current active multisig transaction
+ * @param {string} contractId multisig contract
+ * @returns {IActiveMultisigTx | null}
+ */
+  async function fetchActiveMultisigTx(contractId: string): Promise<IActiveMultisigTx | null> {
```
Comment (peronczyk) [inline src/composables/multisigTransactions.ts:62]: Please fix the indentation.

File: src/composables/multisigTransactions.ts [hunk 9]
```diff
@@ -42,8 +43,44 @@ export function useMultisigTransactions({ store }: UseMultisigTransactionsOption
...
   }
 
-  async function getTransactionByHash(txHash: string) {
-    return fetchJson(`${activeNetwork.value.multisigBackendUrl}/tx/${txHash}`);
+  /**
+ * Get raw proposed transaction from multisig backend
+ * @param {string} txHash
+ * @returns {IRawMultisigTx | null} returns transaction if exists or null
+ */
+  async function getTransactionByHash(txHash: string): Promise<IRawMultisigTx | null> {
```
Comment (peronczyk) [inline src/composables/multisigTransactions.ts:51]: I suggest to rename this function from `get...` to `fetch...` as this is not a getter (getters are not asynchronous).

File: src/composables/multisigAccounts.ts [hunk 10]
```diff
@@ -164,6 +164,7 @@ export function useMultisigAccounts({ store }: UseMultisigAccountsOptions) {
         }
         consensus.expiration_height = Number(consensus.expiration_height);
         consensus.confirmations_required = Number(consensus.confirmations_required);
+        consensus.totalConfirmations = Number(consensus.confirmed_by.length);
```
Comment (peronczyk) [inline src/composables/multisigAccounts.ts:167]: camelCase and snake_case are mixed here. MAybe there is a solution to have one approach here?

File: src/composables/multisigTransactions.ts [hunk 11]
```diff
@@ -42,8 +43,44 @@ export function useMultisigTransactions({ store }: UseMultisigTransactionsOption
     });
   }
 
-  async function getTransactionByHash(txHash: string) {
-    return fetchJson(`${activeNetwork.value.multisigBackendUrl}/tx/${txHash}`);
+  /**
+ * Get raw proposed transaction from multisig backend
+ * @param {string} txHash
+ * @returns {IRawMultisigTx | null} returns transaction if exists or null
```
Comment (peronczyk) [inline src/composables/multisigTransactions.ts:49]: As we are using TypeScript in my opinion there is no need to use DocBlock here.

File: src/composables/multisigTransactions.ts [hunk 12]
```diff
@@ -42,8 +43,44 @@ export function useMultisigTransactions({ store }: UseMultisigTransactionsOption
...
+  async function getTransactionByHash(txHash: string): Promise<IRawMultisigTx | null> {
+    return fetchJson(`${activeNetwork.value.multisigBackendUrl}/tx/${txHash}`)
+      .then((res) => res)
+      .catch(null);
+  }
+
+  /**
+ * Get current active multisig transaction
+ * @param {string} contractId multisig contract
+ * @returns {IActiveMultisigTx | null}
```
Comment (peronczyk) [inline src/composables/multisigTransactions.ts:60]: IDEM

File: src/composables/multisigTransactions.ts (hunk 11 above)
Comment (subhod-i) [reply src/composables/multisigTransactions.ts:49]: DocBlock? you mean to remove {IRawMultisigTx | null} from jsdoc?

File: src/composables/multisigTransactions.ts [hunk 13]
```diff
@@ -2,7 +2,7 @@ import { computed } from '@vue/composition-api';
 import multisigContract from '@aeternity/ga-multisig-contract/SimpleGAMultiSig.aes';
 import type { Store } from 'vuex';
 import {
-  MemoryAccount, Crypto,
+  MemoryAccount, Crypto, TxBuilder,
```
Comment (konradodo) [inline src/composables/multisigTransactions.ts:5]: I suggest to put all items in one line or all in separate.

File: src/composables/multisigAccounts.ts (hunk 10 above)
Comment (CedrikNikita) [inline src/composables/multisigAccounts.ts:167]: We already have a totalConfirmations in the `multisigAccount` object.

Comment (subhod-i) [reply src/composables/multisigAccounts.ts]: Removed it.

Comment (subhod-i) [reply src/composables/multisigAccounts.ts]: There was a solution to use `camelcaseKeysDeep` method but it is breaking the array and bigint values. Hence at the moment we have to keep the same

**Review by peronczyk — APPROVED** [review]

File: src/composables/multisigTransactions.ts [hunk 14]
```diff
@@ -42,8 +38,44 @@ export function useMultisigTransactions({ store }: UseMultisigTransactionsOption
...
+   * @param contractId multisig contract
+   * @returns transaction with consensus details if exists or null
+   */
+  async function fetchActiveMultisigTx(contractId: string): Promise<IActiveMultisigTx | null> {
+    const sdk = await getSdk();
+
+    const gaContractRpc = await sdk.getContractInstance({
+      source: multisigContract,
+      contractAddress: contractId,
+    });
```
Comment (konradodo) [inline src/composables/multisigTransactions.ts:63]:
```suggestion
  async function fetchActiveMultisigTx(contractAddress: string): Promise<IActiveMultisigTx | null> {
    const sdk = await getSdk();

    const gaContractRpc = await sdk.getContractInstance({
      source: multisigContract,
      contractAddress
    });
```

**Review by konradodo — APPROVED** [review]

**Pull request merged**